server/webhook_http_test.go

```diff
@@ -19,6 +20,20 @@ import (
 	"github.com/mattermost/mattermost-server/plugin/plugintest/mock"
 )
 
+func startServer(t *testing.T) *httptest.Server {
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte{})
+	}))
+	l, err := net.Listen("tcp", "localhost:1111")
```

Comment:
2/5 start the test server by default, and then use ts.URL in the tests? The problem with using a fixed port is that it can conflict with other tests.

If you use the dynamic URL, maybe you could re-inject it back into the payload near https://github.com/mattermost/mattermost-plugin-jira/blob/59ded5f9c21a6fbcb2088dc9c4fd39a73751cc1d/server/webhook_http_test.go#L26 ?